Packages/com.unity.inputsystem/InputSystem/Monobehaviors/TrackedPoseDriver.cs

```diff
@@ -0,0 +1,237 @@
+
+        [SerializeField]
+        InputAction m_RotationAction;
+        public InputAction rotationAction { get { return m_RotationAction; } set { m_RotationAction = value; m_RotationBound = false; BindActions();  } }
+      
+        [SerializeField]
+        BasePoseProvider m_PoseProviderComponent = null;
+        public BasePoseProvider poseProviderComponent
+        {
+            get { return m_PoseProviderComponent; }
+            set
+            {
+                m_PoseProviderComponent = value;
+                if (value != null)
+                {
+                    UnbindActions();
+                }
+                else
+                {
+                    BindActions();
+                }
+            }
+        }
+
+        Vector3 m_CurrentPosition = Vector3.zero;
+        Quaternion m_CurrentRotation = Quaternion.identity;
+        bool m_RotationBound = false;
+        bool m_PositionBound = false;
+
+        void BindActions()
+        {
+            if (m_PoseProviderComponent == null)
+            {
+                if (!m_PositionBound && m_PositionAction != null)
+                {
+                    m_PositionAction.performed += OnPositionUpdate;
+                    m_PositionBound = true;
+                    m_PositionAction.Enable();
+                }
+                if (!m_RotationBound && m_RotationAction != null)
+                {
+                    m_RotationAction.performed += OnRotationUpdate;
+                    m_RotationBound = true;
+                    m_RotationAction.Enable();
+                }
+            }
+        }
+
+        void UnbindActions()
+        {
+            if (m_PositionAction != null && m_PositionBound)
+            {
+                m_PositionAction.Disable();
+                m_PositionAction.performed -= OnPositionUpdate;
+                m_PositionBound = false;
+            }
+            if (m_RotationAction != null && m_RotationBound)
+            {
+                m_RotationAction.Disable();
+                m_RotationAction.performed -= OnRotationUpdate;
+                m_RotationBound = false;
+            }
+        }
+
+        void OnPositionUpdate(InputAction.CallbackContext context)
+        {
+            if (m_PositionBound)
+            {
+                var vec3Control = context.control as Vector3Control;
+                if (vec3Control != null)
+                {
+                    m_CurrentPosition = vec3Control.ReadValue();
+                }
+            }
+        }
+
+        void OnRotationUpdate(InputAction.CallbackContext context)
+        {
+            if (m_RotationBound)
+            {
+                var quatControl = context.control as QuaternionControl;
+                if (quatControl != null)
+                {
+                    m_CurrentRotation = quatControl.ReadValue();

```

Comment:
Perform `ReadValue` on context, not on control.

Reply:
ok!